### PR TITLE
Various code view improvements

### LINKED ISSUE #18074: Unable to scroll to see all content in Blame window
### Gitea Version

v1.16-dev

### Git Version

v2.30.1

### Operating System

MacOS ARM1

### How are you running Gitea?

Local build from main

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Description

When viewing a large file in Blame, you can't scroll to the right.

Happening for me in Chrome, but not in Safari. Can scroll in Safari.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by silverwind**
1. Restore missing styles for message close icon
2. Move `code-line-button` so that it does not go off-screen on small viewports
3. Make `code-line-button` look and behave like other buttons
4. Make `code-line-button` work in blame
5. Make the active selection span the whole line, not just the code part
6. Tweak colors, make dark theme code bg darker, make line numbers same color in diff and file view.
7. Move code background to parent, fixing border radius and other problems
8. Enable code wrap in blame
9. Improve blame responsiveness
10. Remove `--color-code-sidebar-bg` in blame, now it uses same background as code
11. Rename `--color-active-line` to `--color-highlight-bg`
12. Add `--color-highlight-bg`
13. Fix button group borders on hover and border-right on last button.

Screenshot 2024-03-23 at 22 34 13
Screenshot 2024-03-23 at 22 34 26

Screenshot 2024-03-23 at 22 34 57
Screenshot 2024-03-23 at 22 34 42

Fixes: https://github.com/go-gitea/gitea/issues/18074

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Comment by yardenshoham** [comment]
Did... Did you just add jQuery code to the codebase?

**Comment by yardenshoham** [comment]
What have I done to deserve this

**Comment by delvh** [comment]
I think the problem is that this file is already using jQuery heavily, so using something else would only be more confusing at the moment.

**Comment by yardenshoham** [comment]
Fine :(

**Comment by silverwind** [comment]
Can rewrite without I guess. The whole function was very hard to follow because of those confusing variable names it has.

**Comment by silverwind** [comment]
I guess this is as much refactor as I'm willing to do right now. Variable names are much clearer now and some jQuery is removed.

**Comment by silverwind** [comment]
More tweaks done. Code sidebar is gone on blame. Two new CSS variables for the highlight added. Blame button muted, background moved to parent, fixing border radius issue.

Screenshot 2024-03-23 at 15 48 00
Screenshot 2024-03-23 at 15 48 14

Screenshot 2024-03-23 at 15 48 46
Screenshot 2024-03-23 at 15 48 28

**Comment by silverwind** [comment]
Unescape button was missing right border in blame view, fixed in https://github.com/go-gitea/gitea/pull/30014/commits/b867329e52568770796d5d7fc2c6ed7666c9b55d so that 2nd and 2nd to last button is considered for border addition:

image

**Comment by silverwind** [comment]
Blame commit links now look like this:

Screenshot 2024-03-23 at 16 09 39

I also added underline styles to base, I think they were lost as a regression from https://github.com/go-gitea/gitea/pull/29980.

**Comment by silverwind** [comment]
Regression with `a:hover` from https://github.com/go-gitea/gitea/pull/29980 confirmed. I cleaned up the styles and also added a devtest example for links:

image

**Comment by silverwind** [comment]
Screenshots updated in OP. Ready to go imho.

**Comment by silverwind** [comment]
Some more minor fixed in blame:

Screenshot 2024-03-23 at 22 58 41

**Comment by silverwind** [comment]
I also enabled code wrapping in blame view now:

Screenshot 2024-03-23 at 23 41 09

File: web_src/css/modules/button.css [hunk 1]
```diff
@@ -61,11 +61,17 @@ It needs some tricks to tweak the left/right borders with active state */
...
+.ui.buttons .button:first-child,
+.ui.buttons .button.gt-hidden:first-child + .button {
   border-left: 1px solid var(--color-light-border);
 }
 
-.ui.buttons .button:last-child {
+.ui.buttons .button:last-child,
+.ui.buttons .button:nth-last-child(2):has(+ .button.gt-hidden) {
   border-right: 1px solid var(--color-light-border);
 }
```
Comment (silverwind) [inline web_src/css/modules/button.css:78]: These changes should fix every remaining issue with the button groups. "color is approximation" really means what it says. The problem is we have lack the color in our variables because it's a mix of transparent colors and in my testing, `color-mix` can not accurately reproduce it.

Comment (silverwind) [reply web_src/css/modules/button.css:78]: Actually, I found a fix so this approximation is no longer needed by just styling all hover styles explicitely with an opaque color.

File: web_src/js/features/repo-code.js [hunk 2]
```diff
@@ -47,17 +53,15 @@ function selectRange($list, $select, $from) {
   };
 
   const updateCopyPermalinkUrl = function(anchor) {
-    if ($copyPermalink.length === 0) {
-      return;
-    }
+    if ($copyPermalink.length === 0) return;
```
Comment (6543) [inline web_src/js/features/repo-code.js]:
```suggestion
    if (!$copyPermalink.length) return;
```

too?

**Comment by silverwind** [comment]
I will try to get `code-line-button` not to hide line number is between 1 and 3 digits, e.g. like GitHub:

image

**Review by 6543 — APPROVED** [review]
lets merge this and itterate on it
